connect/src/routes/tokenBridge/automatic.ts

```diff
@@ -133,18 +125,46 @@ export class AutomaticTokenBridgeRoute<N extends Network> extends AutomaticRoute
     yield* TokenTransfer.track(this.wh, receipt, timeout, this.fromChain, this.toChain);
   }
 
-  private toTransferDetails(
-    amount: bigint,
-    options: AutomaticTokenBridgeRoute.Options,
-  ): TokenTransferDetails {
+  private async stringToBigInt(amount: string = "0"): Promise<bigint> {
+    const { from } = this.request;
+    let decimals = isTokenId(this.request.source)
+      ? await this.wh.getDecimals(from.chain, from.address)
+      : BigInt(this.fromChain.config.nativeTokenDecimals);
+
+    return normalizeAmount(amount, decimals);
+  }
+
+  private async toTransferDetails(params: TransferParams<Op>): Promise<TokenTransferDetails> {
     const { source, from, to } = this.request;
+    const amount = await this.stringToBigInt(params.amount);
+    let options = params.options ?? AutomaticTokenBridgeRoute.getDefaultOptions();
+
+    // Determine nativeGas
+    let nativeGas = 0n;
+
+    // Calculate nativeGas in base units if options.nativeGas isn't 0
+    if (options && options.nativeGas > 0) {
+      const quote = await this.quote(params);
```

Comment:
does this do an infinite recursion since we call toTransferDetails in `quote`?

Reply:
instead of getting a full `quote` we can ask for _just_ the fee from the auto tb?

Reply:
lmfao oops